app/src/main/java/com/hedvig/app/feature/claims/ui/pledge/HonestyPledgeBottomSheet.kt

```diff
@@ -9,16 +9,23 @@ import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import com.hedvig.app.R
 import com.hedvig.app.databinding.BottomSheetHonestyPledgeBinding
 import com.hedvig.app.feature.claims.service.ClaimsTracker
+import com.hedvig.app.feature.claims.ui.ClaimsViewModel
 import com.hedvig.app.feature.embark.ui.EmbarkActivity
+import com.hedvig.app.util.extensions.startClosableChat
 import com.hedvig.app.util.extensions.view.setHapticClickListener
 import com.hedvig.app.util.extensions.viewLifecycleScope
+import com.hedvig.app.util.featureflags.Feature
+import com.hedvig.app.util.featureflags.FeatureManager
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
 import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 
 class HonestyPledgeBottomSheet(
     private val customActivityLaunch: ((Intent) -> Unit)? = null,
 ) : BottomSheetDialogFragment() {
     private val tracker: ClaimsTracker by inject()
+    private val featureManager: FeatureManager by inject()
+    private val claimsViewModel: ClaimsViewModel by sharedViewModel()
```

Comment:
Does HonestyPledgeBottomSheet absolutely need to share this VM? My first instinct would be no, but maybe this is the only way to properly open the claims chat?